tests/python/openassetio/test_imports.py

```diff
@@ -115,3 +121,18 @@ def test_importing_PluginSystemManagerInterfaceFactory_succeeds(self):
 
     def test_importing_PluginSystemPlugin_succeeds(self):
         from openassetio.pluginSystem import PluginSystemPlugin
+
+
+class Test_test_imports:
```

Comment:
[question] Looks all good, I'm just wondering what the rule is for what should be included in this test module.  

At first I thought the intention was to test classes have been hoisted successfully in `__init__.py`s. But the tests here and below break that rule.

Reply:
Originally, the main intent of these tests was to catch cyclic imports etc... so I guess it should be that any individually importable module is importable, but it also doubles up testing C++ hoists now too. So I guess its:

- Any individual python defined module
- Any C++ hoisted module.

Reply:
Can you think of anything else? Ill add a note to such an effect at the top

Reply:
In which case we're missing `test.manager.harness`/`apiComplianceSuite`?  Or is it that these are somewhat "private"?

Reply:
Added notes on coverage [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/534/commits/52d92df5eed34eb32374273fc9e21d5247213b8f), hows that? Added missing tests [https://github.com/OpenAssetIO/OpenAssetIO/pull/534/commits/be3a4e649a7c969296ec27b1515a8187e49e87d0] - somehow I'd totally spaced them.

Reply:
LGTM, thanks.